api/app_sph_lms/api/serializer/course_serializer.py

```diff
@@ -31,20 +32,48 @@ def get_trainee(self, obj):
 
 
 class CourseSerializer(serializers.ModelSerializer):
-    categories = serializers.SerializerMethodField()
+    category = serializers.PrimaryKeyRelatedField(
+        queryset=Category.objects.all(),
+        many=True
+    )
+
     lessons = LessonSerializer(many=True)
 
     class Meta:
         model = Course
         fields = "__all__"
 
-    def get_categories(self, obj):
-        categories = obj.coursecategory_set.all()
-        return [
+    def to_representation(self, instance):
+        representation = super().to_representation(instance)
+        representation['category'] = [
             {
-                "id": category.category.id,
-                "name": category.category.name
-            } for category in categories]
+                'id': category.id,
+                'name': category.name
+            } for category in instance.category.all()
+        ]
+        return representation
+
+    def create(self, validated_data):
+        user = self.context['request'].user
+
+        if not user.is_authenticated or \
+                user.role.title not in ['Trainer', 'Admin']:
+            raise PermissionDenied(
+                "Only authenticated Trainers and Admins can create a course."
+            )
+
+        categories_data = validated_data.pop('category')
+        lessons_data = validated_data.pop('lessons')
+
+        course = Course.objects.create(**validated_data)
+
+        for index, lesson_data in enumerate(lessons_data):
+            lesson_data['course'] = course
+            lesson_data['order'] = index + 1
```

Comment:
There is a specific value for order that is coming from FE, so please set that order for order of lesson, not the index from for loop iteration, it might cause a bug